### ISSUE TITLE
Updating how we run integ tests

### ISSUE THREAD

**Issue by amitgalitz** [opensearch-project/anomaly-detection#451]
**Is your feature request related to a problem? Please describe.**
AD has experienced multiple flaky tests that are occurring at a rate that slows down development. AD’s Github workflows takes around 1 hour to complete, these are executed before merging and on each new commit that is made on an open PR. 

AD and other OpenSearch repos are relaying on Github’s runners that are quite small (2 x86_64 CPU cores and 7 GB RAM). This seems to be the main reason for many of AD’s flaky tests as there isn’t enough hardware processing power to consistently pass all our tests as we create indices, ingest data, start detectors, updates detectors and etc.

I propose that we consider a few possible solutions to at least reduce the amount of times we have to re-start our Github workflows or how long they take.

**Describe the solution you'd like**
Solution options (could be a variation of multiple options):

1. Explore the use of self hosted GitHub runners where we can take greater control of what machines are used to run our CI.
    1. This has a few downsides like increased cost, managing our own infrastructure and security issues (OpenSearch core currently isn’t pursuing this)
2. Wait for public Jenkins (potential release April 6th as mentioned in this issue’s comments) and AD CI too it, we can have an increased instance size on Jenkins to reduce the failing tests. Need to do more research on how this exactly would work and in terms of our GitHub commit process right now.
3. Further divide our workflow into smaller tasks. For example we can potentially separate the `Build and Run Tests` workflow into smaller tasks. We can also investigate separating out specific integ-tests into different workflows. Reducing how long each tasks runs has proven to reduce flakiness on OpenSearch core testing. *(This can be done anyways regardless of other options)*
    1. Additionally we can investigate if more workflows can run in parallel to reduce CI time (not necessarily flakiness)
4. Separate which specific integration tests we are running for each new commit/PR, when we merge to main and when infra team does validation testing. 
    1. Full integration testing could still be done on a new PR but for example run only a set of the integration tests when infra team does validation testing
5. https://blog.gradle.org/gradle-flaky-test-retry-plugin
